### ISSUE TITLE
compiler does not give error if we pass invalid parameters to override abstract method in derived class

### ISSUE THREAD

**Issue by SatishChauhanKD** [microsoft/TypeScript#39598]
**TypeScript Version:** 

tsc --version
Version 3.9.6

tsc --showconfig
"strict": true,
"noImplicitAny": true,
"strictNullChecks": true,
"strictFunctionTypes": true,

**Code**

```ts
abstract class Base {
  public name: string;
  public abstract greet(message: string): string;

  constructor(name: string) {
    this.name = name;
  }
}

class Person extends Base {
  public greet(): string {
    return this.name;
  }

  constructor(name: string) {
    super(name);
  }
}

const p = new Person("Bob");
console.log(p.greet());
```

**Expected behavior:**

please note "strictFunctionTypes": true is set in tsconfig.json
no parameter is pass to greet() in Person class, compiler should not compile code.

**Actual behavior:**
compiler compile code

**Comment by MartinJohns** [comment]
Please respect the [issue template for bug reports](https://raw.githubusercontent.com/microsoft/TypeScript/master/.github/ISSUE_TEMPLATE/Bug_report.md).

**Comment by MartinJohns** [comment]
[Why are functions with fewer parameters assignable to functions that take more parameters?](https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters)

If your implementation can work with fewer parameters, then this is absolutely fine and acceptable.
